### ISSUE TITLE
Permission for the home directory: panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by volker-schukai** [dolthub/dolt#8250]
```
[dolt	] panic: runtime error: invalid memory address or nil pointer dereference
[dolt	] [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x23e62d8]
[dolt	] 
[dolt	] goroutine 1 [running]:
[dolt	] github.com/dolthub/dolt/go/libraries/doltcore/env.(*DoltCliConfig).GetStringOrDefault(...)
[dolt	] 	github.com/dolthub/dolt/go/libraries/doltcore/env/config.go:154
[dolt	] main.runMain()
[dolt	] 	github.com/dolthub/dolt/go/cmd/dolt/dolt.go:477 +0x20b8
[dolt	] main.main()
[dolt	] 	github.com/dolthub/dolt/go/cmd/dolt/dolt.go:247 +0x13
```

Version: 1.38.0
NIXOS: 24.05
dockerTools

I have been able to narrow down the problem to the following:

- dolt is running as user x (docker container)
- the home directory of x exists, but x has no rights to write or read in the home directory.
- If I change the rights (x), it works.
- in the other case, panic. 
 
Of course, this could just be a correlation.

With the right environment, the error does not seem to occur, so it does not seem 
to be a tragic bug, but a panic is never really good.

**Comment by fulghum** [comment]
Thanks for taking the time to report this one @volker-schukai 🙏 

I'm able to repro the issue by running `sudo -u nobody dolt version`. I think there are likely a few codepaths that could trigger this, so I'll explore a bit more and get a change out to turn this into a better error message.

**Comment by fulghum** [comment]
We've released [Dolt 1.42.11](https://github.com/dolthub/dolt/releases/tag/v1.42.11) with a fix for this issue. 

The root cause here was that our code that validates the global configuration directory wasn't getting run early enough and some other code had been added before the validation and tried to use the global configuration. I moved up that validation/error-checking code and also added some news tests for global config directories without execute permission and without read/write permission. 

Thanks for taking the time to let us know about this problem. Let us know if there's anything else we can do to help you build with Dolt!
